### PR TITLE
Feature/tags

### PR DESCRIPTION
**Pull request by fillup**

### CODE REVIEW

File: db/common.go [hunk 1]
```diff
@@ -37,25 +37,90 @@ func GetItem(tableName, attrName, attrValue string, returnObj interface{}) error
...
+	}
+
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func scanTable(tableName string) ([]map[string]*dynamodb.AttributeValue, error) {
```
Comment (Baggerone) [inline db/common.go]: Do you want it to use tableAlias here as well?

File: db/common.go [hunk 2]
```diff
@@ -37,25 +37,90 @@ func GetItem(tableName, attrName, attrValue string, returnObj interface{}) error
...
+	}
+
+	return results, nil
+}
+
+func ListTags() ([]domain.Tag, error) {
+
+	var list []domain.Tag
+
+	items, err := scanTable(domain.GetDbTableName(domain.TagTable))
```
Comment (Baggerone) [inline db/common.go]: If you switch to tableAlias above, you will need to change this param as well.

File: tag/main.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	case "DELETE":
+		return deleteTag(req)
+	default:
+		return domain.ClientError(http.StatusMethodNotAllowed, http.StatusText(http.StatusMethodNotAllowed))
+	}
+}
+
+func viewTag(req events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
+	var tag domain.Tag
+	err := db.GetItem(domain.TagTable, "name", req.PathParameters["name"], &tag)
```
Comment (Baggerone) [inline tag/main.go]: Is it worth checking that the "name" param isn't empty? (may also apply to deleteTag.)
Or, do the generic functions need to do anything to validate the param value?

File: db/common.go (hunk 1 above)
Comment (Baggerone) [inline db/common.go]: Can this be exported (ScanTable)?

Comment (fillup) [reply db/common.go]: Yes I'll do that

File: db/common.go (hunk 2 above)
Comment (fillup) [reply db/common.go]: will do

File: tag/main.go (hunk 3 above)
Comment (fillup) [reply tag/main.go]: it should only get to this method if it is present due to routing on api gateway

**Pull request merged**